cpp/src/arrow/compute/kernels/scalar_arithmetic.cc

```diff
@@ -186,6 +187,42 @@ struct MultiplyChecked {
   }
 };
 
+struct Divide {
+  template <typename T, typename Arg0, typename Arg1>
+  static constexpr enable_if_floating_point<T> Call(KernelContext*, Arg0 left,
+                                                    Arg1 right) {
+    return left / right;
+  }
+
+  template <typename T, typename Arg0, typename Arg1>
+  static enable_if_integer<T> Call(KernelContext* ctx, Arg0 left, Arg1 right) {
+    if (ARROW_PREDICT_FALSE(right == 0)) {
+      ctx->SetStatus(Status::Invalid("overflow"));
+      return 0;
+    }
+    return left / right;
+  }
+};
+
+struct DivideChecked {
+  template <typename T, typename Arg0, typename Arg1>
+  static enable_if_integer<T> Call(KernelContext* ctx, Arg0 left, Arg1 right) {
+    static_assert(std::is_same<T, Arg0>::value && std::is_same<T, Arg1>::value, "");
+    T result;
+    if (ARROW_PREDICT_FALSE(DivideWithOverflow(left, right, &result))) {
+      ctx->SetStatus(Status::Invalid("overflow"));
```

Comment:
```suggestion
      if (right == 0) {
        ctx->SetStatus(Status::Invalid("divide by zero"));
      } else {
        ctx->SetStatus(Status::Invalid("overflow"));
      }
```

Reply:
Accepted. Thanks for the good suggestion.